### PR TITLE
story-angular-generic-south-backend

### PR DESCRIPTION
**Pull request by burgerni10**
ON TOP OF #2147

### CODE REVIEW

File: backend/model/south-connector.model.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
+/**
+ * DTO for South connectors
+ */
+export interface SouthConnectorDTO {
+  id: string;
+  name: string;
+  type: string;
+  description: string;
+  enabled: boolean;
+  settings: object;
```
Comment (marouanehassanioptimistik) [inline backend/model/south-connector.model.ts:10]: this doesn't look too good to me, we should have types for settings with inheritance

File: backend/model/south-connector.model.ts [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+/**
+ * Command DTO for South connector
+ */
+export interface SouthConnectorCommandDTO {
+  name: string;
+  type: string;
+  description: string;
+  enabled: boolean;
+  settings: object;
```
Comment (marouanehassanioptimistik) [inline backend/model/south-connector.model.ts:21]: same here

**Review by marouanehassanioptimistik — CHANGES_REQUESTED** [review]

File: backend/model/south-connector.model.ts (hunk 1 above)
Comment (burgerni10) [reply backend/model/south-connector.model.ts:10]: I don't think it will be compatible with a plugin approach ?

Comment (burgerni10) [reply backend/model/south-connector.model.ts:10]: I suggest to keep it like that for now, and manage that in a future PR when I'll change the schema on the backend

Comment (marouanehassanioptimistik) [reply backend/model/south-connector.model.ts:10]: we should make it compatible i don't see the reason it shouldn't be the plugins should import our types and extend accordingly the base class

Comment (marouanehassanioptimistik) [reply backend/model/south-connector.model.ts:10]: if we don't have a model for settings then we have no type safety for south connectors which doesn't seem to good for me. ok to take that into account in a subsequent pr.

**Pull request merged**